WMF Framework/Event Platform Client/EPCNetworkManager.swift

```diff
@@ -0,0 +1,135 @@
+
+import Foundation
+
+class EPCNetworkManager: EPCNetworkManaging {
+
+    private let session: Session
+
+    /**
+     * Serial dispatch queue that enables working with properties in a thread-safe
+     * way
+     */
+    private let queue = DispatchQueue(label: "EPCNetworkManager-" + UUID().uuidString)
+    /**
+     * Holds events that have been scheduled for POSTing
+     */
+    private var outputBuffer: [(url: URL, body: NSDictionary)] = []
+    
+    init(session: Session = Session.shared) {
+        self.session = session
+    }
+
+    func schedulePost(url: URL, body: NSDictionary) {
+        appendPostToOutputBuffer((url: url, body: body))
+    }
+    
+    func httpDownload(url: URL, completion: @escaping (Data?) -> Void) {
+        httpDownload(url: url, retriesRemaining: 5, attemptDelay: 2, completion: completion)
+    }
+
+    /**
+     * Download data over HTTP with capacity for retries
+     * - Parameters:
+     *   - url: where to request data from
+     *   - retriesRemaining: maximum number of retries allowed for this download
+     *     operation
+     *   - attemptDelay: time between each retry
+     *   - completion: what to do with the downloaded data
+     *
+     * The operation will be performed up to `retriesRemaining + 1` times.
+     */
+    private func httpDownload(url: URL, retriesRemaining: Int, attemptDelay: TimeInterval, completion: @escaping (Data?) -> Void) {
```

Comment:
Because this is only ever used to fetch stream configurations, and only ever receives one completion handler, I would suggest we name this `fetchStreamConfiguration` and investigate inlining it in the main library class.
```swift
private func fetchStreamConfiguration(retries: Int, retryDelay: Date) {
    networkManager.httpGet(url: streamConfigurationServiceURI, completion: { 
        (data, response, error) in 
            
        if let res = response, res.statusCode == 200 {
            if let dat = data {
                loadStreamConfiguration(dat)
            }
        } else {
            if retries > 0 {
                dispatchOnMainQueueAfterDelayInSeconds(retryDelay) {
                    fetchStreamConfiguration(retries: retries-1, retryDelay: retryDelay)
                }
            } else {
                    /* do the thing that you want to do when you can't load stream configuration */
            }
        }
    })
}
```
Here, `loadStreamConfiguration` would be the completion handler currently in `configure()`. `configure` then basically becomes the code above. The other thing you need is the method `httpGet`, which could look like:
```swift
private func httpGet(url: String, completion: @escaping (Data?, HTTPURLResponse?, Error?) -> Void) {
    let request = URLRequest( url: URL(string: url) )
    request.httpMethod = "GET"

    let task = session.dataTask(with: request, completionHandler: completion) 
    task?.resume()
}
```